app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -141,18 +143,49 @@ function ContainerTopologyCtrl($scope, $http, $interval, $location, topologyServ
       return self.dblclick(d);
     });
 
+    added.append("image")
+      .attr("xlink:href", function (d) {
+        switch(d.item.kind) {
+          case 'ContainerManager':
+              return icons[d.item.display_kind];
+            //return 'vendor-'+ d.item.display_kind + ".svg";
+          default:
+            return null;
+        }
+      })
+      .attr("height", function(d) {
+         return 40;
+      })
+      .attr("width", function(d) {
+         return 40;
+      })
+      .attr("y", function(d) {
+        return self.getDimensions(d).y;
+      })
+      .attr("x", function(d) {
+        return self.getDimensions(d).x;
+      })
+      .on("contextmenu", function(d){
+        self.contextMenu(this, d);
+      });
+
     added.append("text")
-        .text(function(d) {
-          return self.getIcon(d);
-        })
-      .attr('class', function(d) {
+      .text(function(d) {
           switch(d.item.kind) {
             case 'ContainerManager':
-              return 'icon '+ d.item.display_kind;
+              return null;
             default:
-              return 'icon';
+              return self.getIcon(d);
           }
-        })
+      })
+      .attr('class', function(d) {
+        switch(d.item.kind) {
+          case 'ContainerManager':
```

Comment:
This look a bit against what we are trying to do in general.
I mean we want the code to query properties and capabilities of entities not types of entities. There should be no need to ask for 'ContainerManager' and such.